### ISSUE TITLE
XMLHttpRequest polyfill prevents importing as a module

### ISSUE THREAD

**Issue by ssorallen** [mudcube/MIDI.js#110]
**Q:** Do you copy/paste new versions of "DOMLoader.XMLHttp.js"? Where should this fix actually go?
### TL;DR

DOMLoader.XMLHttp.js should polyfill `XMLHttpRequest` without expecting to be executed in the global context. This will allow MIDI.js to be shimmed and included by module loaders such as Webpack and Browserify.
### El Problema

JS bundlers like Webpack and Browserify can shim libraries that are not built as modules, and they do so by wrapping the library in IIFEs. When attempting to shim MIDI.js using Webpack, MIDI.js always throws an exception:

```
Uncaught TypeError: undefined is not a function
```

This is due to the way [DOMLoader.XMLHttp.js](https://github.com/mudcube/MIDI.js/blob/master/js/Window/DOMLoader.XMLHttp.js) attempts to polyfill `XMLHttpRequest`. It is written in a way that _must_ be executed in the global context. This statement is the cause:

```
if (typeof (XMLHttpRequest) === "undefined") {
    var XMLHttpRequest;
        ...
}
```

When that is executed in the global scope the `var XMLHttpRequest` simply redeclares the already-defined `XMLHttpRequest` in most browsers, which is an innocuous action in JavaScript. Redeclaring a variable does not affect the value of the variable. Here's some console action to demonstrate:

```
> XMLHttpRequest
function XMLHttpRequest() { [native code] }
> var XMLHttpRequest
undefined
> XMLHttpRequest
function XMLHttpRequest() { [native code] }
```

However, when MIDI.js is wrapped in an IIFE the `var` declaration is hoisted to the top of the function and the `if (typeof (XMLHttpRequest) === "undefined")` check passes because there is now a function-local, undefined variable named `XMLHttpRequest`. More console to demonstrate:

```
> (function() {
    var XMLHttpRequest;
    console.log(XMLHttpRequest);
  })();
undefined
undefined
```
### The Fix

DOMLoader.XMLHttp.js should create a local `XMLHttpRequest` that is initialized to the global value. That will allow it to be executed in any context and still correctly poyfill `XMLHttpRequest` for older versions of Internet Explorer.

```
var XMLHttpRequest = XMLHttpRequest;
if (typeof (XMLHttpRequest) === "undefined") {
    (function () { // find equivalent for IE
        ...
        XMLHttpRequest = factories[i];
    })();
}
```
